chrome/content/zotero/collectionTree.jsx

```diff
@@ -1567,6 +1568,21 @@ var CollectionTree = class CollectionTree extends LibraryTree {
 		// Directly on a row
 		if (orient === 0) {
 			if (!treeRow.editable) {
+				// Can drop collections, searches and items into trash
+				if (treeRow.isTrash()) {
+					if (dataType == 'zotero/item') {
+						var items = Zotero.Items.get(data);
+						let allInSameLibrary = !items.find(item => item.libraryID != treeRow.ref.libraryID);
+						// Cannot drop items from another library
+						return allInSameLibrary;
+					}
+					if (dataType == 'zotero/search' || dataType == 'zotero/collection') {
+						// Can only drop collections and searches into trash of their library
+						let id = data[0];
+						let obj = dataType == 'zotero/collection' ? Zotero.Collections.get(id) : Zotero.Searches.get(id);
+						return treeRow.ref.libraryID == obj.libraryID;
+					}
```

Comment:
Do these need to be different? Isn't it the same check?

Reply:
Apologies, I'm not sure what exactly you mean... 

In this `canDropCheck` bit, `treeRow` is the trash row, and a given `id` from `data` is the id of whatever object is being dragged. So we want to make sure that the search or collection being dragged is allowed to be dropped into the trash of only its own library. And I'm checking against the `dataType` to decide if it's a `Zotero.Collection` or `Zotero.Search` where the actual object should be fetched from.

Alternatively, it could have been something like
```
if (dataType == 'zotero/collection') return Zotero.Collections.get(id).libraryID == treeRow.ref.libraryID;
if (dataType == 'zotero/search') return Zotero.Searches(id).libraryID == treeRow.ref.libraryID;
```

Reply:
I mean why do items and collections/searches use different logic?

Reply:
Oh, gotcha. Yeah, it really is the same logic. I think I wanted to avoid too many `if-else` statements which did lead to this part looking off. I refactored it to just fetch the objects based on the `dataType` and compute `allInSameLibrary` for that array. That should make more sense now!